tools/generators/files_and_groups/src/Generator/CalculateTargetFilesPartial.swift

```diff
@@ -0,0 +1,73 @@
+import PBXProj
+
+extension Generator {
+    struct CalculateTargetFilesPartial {
+        private let callable: Callable
+
+        /// - Parameters:
+        ///   - callable: The function that will be called in
+        ///     `callAsFunction()`.
+        init(callable: @escaping Callable = Self.defaultCallable) {
+            self.callable = callable
+        }
+
+        /// Calculates the `PBXBuildFile`s and product `PBXFileReference`s
+        /// partial.
+        func callAsFunction(objects: [TargetFileObject]) -> String {
+            return callable(/*objects:*/ objects)
+        }
+    }
+}
+
+// MARK: - CalculateTargetFilesPartial.Callable
+
+extension Generator.CalculateTargetFilesPartial {
+    typealias Callable = (
+        _ objects: [TargetFileObject]
+    ) -> String
+
+    static func defaultCallable(
+        objects: [TargetFileObject]
+    ) -> String {
+        var buildFiles: [Object] = []
+        var productIdentifiers: [(
+            subIdentifier: Identifiers.BuildFiles.SubIdentifier,
+            identifier: String
+        )] = []
+        for targetFileObject in objects {
+            switch targetFileObject {
+            case .buildFile(let object):
+                buildFiles.append(object)
+            case .product(let subIdentifier, let identifier):
+                productIdentifiers.append((subIdentifier, identifier))
+            }
+        }
+
+        // FIXME: Sort `productIdentifiers`, right? May need more info (like SubIdentifiers here)
```

Comment:
Are you planning to address this `FIXME` in this PR?

Reply:
I do see a `productIdentifiers.sorted` below though

Reply:
Forgot to remove the fixme.